### PR TITLE
LPS-111636 Implement default addresses for accounts

### PR DESCRIPTION
**Pull request by pei-jung**
/cc @alee8888

### CODE REVIEW

**Comment by pei-jung** [comment]
ci:test:sf

**Comment by pei-jung** [comment]
ci:test:relevant

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+    AccountEntryDisplay accountEntryDisplay = (AccountEntryDisplay)request.getAttribute(AccountWebKeys.ACCOUNT_ENTRY_DISPLAY);
```
Comment (drewbrokke) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp]: nit: We should unindent this line

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        </clay:content-col>
+    </clay:content-row>
+
+    <div class="form-group-autofit">
+        <div class="form-group-item">
+            <div class="sheet-text">
+                <liferay-ui:message key="billing" />
+                <clay:icon symbol="credit-card" />
+
+                <liferay-text-localizer:address-display address="<%= %>" />
```
Comment (drewbrokke) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp]: Is this correct? `<%= %>`

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            </div>
+        </div>
+        <div class="form-group-item">
+            <div class="sheet-text">
+                <liferay-ui:message key="shipping" />
+                <clay:icon symbol="truck" />
+            </div>
+        </div>
+    </div>
+</clay:sheet-section>
```
Comment (drewbrokke) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/default_addresses.jsp]: I'm guessing this file was not supposed to be added in this commit. I don't think it's worth closing the pull over though.

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/js/DefaultAddresses.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+		'.modify-link',
+		onClick
+	);
+
+	return {
+		dispose() {
+			clickDelegate.dispose();
+		},
+	};
+}
```
Comment (drewbrokke) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/js/DefaultAddresses.js:89]: @pei-jung Did you work off of an example of this kind of JS component? I like that there's an option to use something besides a class.

File: modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/select_default_address.jsp [hunk 5]
```diff
@@ -17,6 +17,19 @@
...
+
+String type = ParamUtil.getString(request, "type");
+
+if (Objects.equals("billing", type)) {
+	defaultAddressId = accountEntryDisplay.getDefaultBillingAddressId();
+}
+else if (Objects.equals("shipping", type)) {
+	defaultAddressId = accountEntryDisplay.getDefaultShippingAddressId();
+}
+
```
Comment (drewbrokke) [inline modules/apps/account/account-admin-web/src/main/resources/META-INF/resources/account_entries_admin/account_entry/select_default_address.jsp:32]: Do we also need to account for a type is "billing-and-shipping"?

**Review by drewbrokke — COMMENTED** [review]
Great job on this @pei-jung. Only minor comments on this one, I'll let you decide if you want to forward it or make the changes.

**Comment by drewbrokke** [comment]
ci:forward

**Pull request closed without merging**